### PR TITLE
Pass OpenFlags for OpenFileOp

### PR DESCRIPTION
**Pull request by xinmeigui-db**
Pass OpenFlags for OpenFileOp, so that we could have custom logic based on the operation.

### CODE REVIEW

File: fuseops/ops.go [hunk 1]
```diff
@@ -646,6 +648,8 @@ type OpenFileOp struct {
 	// advance, for example, because contents are generated on the fly.
 	UseDirectIO bool
 
+	OpenFlags *fusekernel.OpenFlags
```
Comment (xinmeigui-db) [inline fuseops/ops.go:651]: I made this a pointer, so we don't default to `OpenReadOnly` (which is 0). Let me know if you have feedback.

**Review by stapelberg — APPROVED** [review]

**Comment by stapelberg** [comment]
Thanks!

**Comment by xinmeigui-db** [comment]
@stapelberg Thanks for the review! I was actually planning to make this a non-pointer, given it should always be set. Also was hoping to add more testing. If those sound good to you, I will open another PR with those changes.

**Comment by stapelberg** [comment]
sure

**Comment by xinmeigui-db** [comment]
@stapelberg I created https://github.com/jacobsa/fuse/pull/129. Please take a look when you get a chance, thanks!

**Pull request merged**